libs/community/langchain_community/chat_message_histories/redis.py

```diff
@@ -49,7 +51,7 @@ def key(self) -> str:
     @property
     def messages(self) -> List[BaseMessage]:
         """Retrieve the messages from Redis"""
-        _items = self.redis_client.lrange(self.key, 0, -1)
+        _items = self.redis_client.lrange(self.key, -self.k*2, -1)
```

Comment:
nit: why not `-self.k`, to provide more flexibility? My instinct is for `k` to correspond to messages instead of conversational turns. but just my opinion.

Reply:
In my opinion,`k` represents `k` pairs of HumanMessage and AIMessage, and `k*2` is the number of BaseMessage.For the next conversation, History will be more complete with `k` message pairs than with `k ` messages.

Reply:
> nit: why not `-self.k`, to provide more flexibility? My instinct is for `k` to correspond to messages instead of conversational turns. but just my opinion.
In my opinion,`k` represents `k` pairs of HumanMessage and AIMessage, and `k*2` is the number of BaseMessage.For the next conversation, History will be more complete with `k` message pairs than with `k` messages.

Reply:
This can get unwieldy outside a simple chatbot use-case. For example, if a model calls a tool, there are additional tool messages that would be added to the history. `self.k * 2` prevents retrieval of an odd number of messages. `self.k` has no restrictions.